### PR TITLE
Explain the new directory structure with composer-installer 5

### PR DESCRIPTION
**Pull request by linawolf**
Resolves https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/203

I will make follow ups for
- Migration
- Situation in v11

releases: main

### CODE REVIEW

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 1]
```diff
@@ -96,20 +115,10 @@ So all files like CSS, JavaScript, Icons, Fonts, Images, etc. of extensions
...
-    The Composer plugin `typo3/cms-composer-installers` in version 4+ was created
-    for TYPO3 version 12 and backported for default but **optional usage**
-    in TYPO3 version 11. Therefore the version has to be explicitely set (decreased)
-    if the classical directory structure shall be used:
-
-.. code-block:: none
-
-   "typo3/cms-composer-installers": "^3.1",
+..  note::
+    In TYPO3 v12 using the `typo3/cms-composer-installers` in version
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:119]: I would rephrase it, as it is installed automatically, the sentence now can be understood as the user has to/can do something.
Maybe: "TYPO3 v12 requires typo3/cms-composer-installers in version 5."

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 2]
```diff
@@ -178,8 +187,9 @@ for temporary files of extensions and TYPO3 components.
 :file:`public/typo3temp/assets/`
 """"""""""""""""""""""""""""""""
 
-Directory for temporary files that should be public available
-(e.g. generated images).
+Directory :file:`typo3temp/assets/` contains temporary files that should be
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:190]:
```suggestion
The directory :file:`typo3temp/assets/` contains temporary files that should be
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 3]
```diff
@@ -80,9 +80,28 @@ to be used:
...
+
+This folder contains the main entry script :file:`index.php` created by Composer
+and might contain publicly available files like a :file:`robots.txt` and
+files needed for the server configuration like a :file:`.htaccess`.
+
+If required, this directory can be renamed by setting `extra > typo3/cms > web-dir`
+in the composer.json, for example to :file:`web`:
+
+..  code-block:: json
+    :caption: composer.json
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:94]:
```suggestion
    :caption: composer.json

```

The snippet is not rendered, probably because of missing empty line after caption.

File: Documentation/Configuration/Typo3ConfVars/FE.rst [hunk 4]
```diff
@@ -494,12 +494,16 @@ additionalAbsRefPrefixDirectories
...
 
-   :type: text
-   :Default: ''
+    :type: text
+    :Default: ''
+
+    Enter additional directories to be prepended with absRefPrefix.
+    Directories must be comma-separated. TYPO3 already prepends the following
+    directories :file:`public/_assets/`, :file:`public/typo3temp/` and all
+    local storages including :file:`public/fileadmin`.
```
Comment (brotkrueml) [inline Documentation/Configuration/Typo3ConfVars/FE.rst:503]:
```suggestion
    local storages including :file:`public/fileadmin/`.
```

File: Documentation/Configuration/Typo3ConfVars/FE.rst [hunk 5]
```diff
@@ -494,12 +494,16 @@ additionalAbsRefPrefixDirectories
...
 
-   Enter additional directories to be prepended with absRefPrefix.
-   Directories must be comma-separated. TYPO3 already prepends the following
-   directories typo3/, typo3temp/, typo3conf/ext/ and all local storages
+    In legacy installations without Composer :file:`typo3conf/ext`
+    and :file:`typo3` are also prefixed.
```
Comment (brotkrueml) [inline Documentation/Configuration/Typo3ConfVars/FE.rst:506]:
```suggestion
    and :file:`typo3/` are also prefixed.
```

File: Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst [hunk 6]
```diff
@@ -1,34 +1,43 @@
...
-System and Local Extensions
-===============================
+=========================================
+System, third-party and custom extensions
+=========================================
 
-The files for an extension are located in a folder named by the
-*extension key*. The location of this folder can be either inside
-`typo3/sysext/` or `typo3conf/ext/`.
+The files for an extension are installed into a folder named :file:`vendor`
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst:9]:
```suggestion
The files for an extension are installed into a folder named :file:`vendor/`
```

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 7]
```diff
@@ -96,20 +116,10 @@ So all files like CSS, JavaScript, Icons, Fonts, Images, etc. of extensions
...
-    for TYPO3 version 12 and backported for default but **optional usage**
-    in TYPO3 version 11. Therefore the version has to be explicitely set (decreased)
-    if the classical directory structure shall be used:
-
-.. code-block:: none
-
-   "typo3/cms-composer-installers": "^3.1",
+..  note::
+    In TYPO3 v12 using the `typo3/cms-composer-installers` in version
+    5 is mandatory. Therefore the publicly available files provided by
```
Comment (linawolf) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:121]:
```suggestion
    TYPO3 v12 requires `typo3/cms-composer-installers` in version
    5. Therefore the publicly available files provided by
```

File: Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst [hunk 8]
```diff
@@ -37,19 +46,16 @@ extension.
...
+        "name": "typo3/cms-core",
+        "type": "typo3-cms-framework",
+        "...": "..."
+    }
 
-Local extensions take precedence which means that if an extension
-exists both in :file:`typo3conf/ext/` and :file:`typo3/sysext/` the one in :file:`typo3conf/ext/`
-is loaded. This means that extensions are loaded in the
-order of priority local-system.
+In legacy installations they are installed into
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst:61]: It is mentioned above, should we repeat it?

[the remark this answers is not in the document]
Comment (brotkrueml) [reply Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst:61]: Section is about system extensions, you mention legacy, you should also mention composer. Not everyone reads from top to bottom but jumps to the according section.

**Review by brotkrueml — APPROVED** [review]
